packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -0,0 +1,672 @@
+	): Promise<void> {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.Set,
+		);
+
+		const cc = new BarrierOperatorCCSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			targetState,
+		});
+		await this.driver.sendCommand(cc, this.commandOptions);
+	}
+
+	public async getSignalingCapabilities(): Promise<
+		readonly SubsystemType[] | undefined
+	> {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.SignalingCapabilitiesGet,
+		);
+
+		const cc = new BarrierOperatorCCSignalingCapabilitiesGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = await this.driver.sendCommand<BarrierOperatorCCSignalingCapabilitiesReport>(
+			cc,
+			this.commandOptions,
+		);
+		return response?.supportedSubsystemTypes;
+	}
+
+	public async getEventSignaling(
+		subsystemType: SubsystemType,
+	): Promise<SubsystemState | undefined> {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.EventSignalingGet,
+		);
+
+		const cc = new BarrierOperatorCCEventSignalingGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			subsystemType,
+		});
+		const response = await this.driver.sendCommand<BarrierOperatorCCEventSignalingReport>(
+			cc,
+			this.commandOptions,
+		);
+		return response?.subsystemState;
+	}
+
+	public async setEventSignaling(
+		subsystemType: SubsystemType,
+		subsystemState: SubsystemState,
+	): Promise<void> {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.EventSignalingSet,
+		);
+
+		const cc = new BarrierOperatorCCEventSignalingSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			subsystemType,
+			subsystemState,
+		});
+
+		await this.driver.sendCommand(cc, this.commandOptions);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property, propertyKey },
+		value,
+	): Promise<void> => {
+		if (property === "targetState") {
+			if (typeof value !== "number") {
+				throwWrongValueType(
+					this.ccId,
+					property,
+					"number",
+					typeof value,
+				);
+			}
+
+			const targetValue =
+				value === BarrierState.Closed
+					? BarrierState.Closed
+					: BarrierState.Open;
+			await this.set(targetValue);
+
+			// Verify the change after a delay
+			if (this.isSinglecast()) {
+				this.schedulePoll({ property });
+			}
+		} else if (property === "signalingState") {
+			if (propertyKey == undefined) {
+				throwMissingPropertyKey(this.ccId, property);
+			} else if (typeof propertyKey !== "number") {
+				throwUnsupportedPropertyKey(this.ccId, property, propertyKey);
+			}
+			if (typeof value !== "number") {
+				throwWrongValueType(
+					this.ccId,
+					property,
+					"number",
+					typeof value,
+				);
+			}
+			await this.setEventSignaling(propertyKey, value);
```

Comment:
As mentioned above, this works fine using a `signalingState` property that is marked as read-only.

I hate to be the one to suggest it, but should this be refreshed? Or should the application assume it is set, or request its own refresh?

Reply:
signalingState is now writable. As with many other values, I've decided not to verify with a poll. If setting the value does not fail, applications should assume that it worked. If this is **really, really** critical somehow, there's always the option to poll on demand.